### ISSUE TITLE
Formatting in `@AfterTemplate` is ignored, leading to huge single-line statements after the migration

### ISSUE THREAD

**Issue by Philzen** [openrewrite/rewrite-templating#92]
## What version of OpenRewrite are you using?

I am using

- OpenRewrite 8.27.1
- rewrite-templating:1.9.2
- error_prone_core:2.28.0

## How are you running OpenRewrite?

I am using the Maven plugin, and my project is a single module project.
```xml
<plugin>
  <groupId>org.openrewrite.maven</groupId>
  <artifactId>rewrite-maven-plugin</artifactId>
  <version>1.2.3</version>
  <configuration>
    ... 
  </configuration>
</plugin>
```

## What is the smallest, simplest way to reproduce the problem?

I have the following refaster recipe:

```java
    public static class MigrateAssertEqualsNoOrderIteratorWithMessage {

        @BeforeTemplate void before(Iterator<?> actual, Iterator<?> expected, String message) {
            Assert.assertEqualsNoOrder(actual, expected, message);
        }

        @AfterTemplate void after(Iterator<?> actual, Iterator<?> expected, String message) {
            Assertions.assertArrayEquals(
                StreamSupport.stream(Spliterators.spliteratorUnknownSize(expected, 0), false).sorted().toArray(),
                StreamSupport.stream(Spliterators.spliteratorUnknownSize(actual, 0), false).sorted().toArray(),
                message
            );
        }
    }
```

(which is already breaking the 100 character barrier, but gives a better trade-off regarding cognitive complexity)

## What did you expect to see?

The migrated code should keep the formatting.

## What did you see instead?

It creates this monster:

```java
    Assertions.assertArrayEquals(StreamSupport.stream(Spliterators.spliteratorUnknownSize(expected, 0), false).sorted().toArray(), StreamSupport.stream(Spliterators.spliteratorUnknownSize(actual, 0), false).sorted().toArray(), "Should contain the same elements");

```

I can see that the generated code has ignored the formatting, the JavaTemplate string has no line breaks or indents whatsoever:

```
    @NonNullApi
    public static class MigrateAssertEqualsNoOrderIteratorWithMessageRecipe extends Recipe {
    // …
                final JavaTemplate after = JavaTemplate
                        .builder("org.junit.jupiter.api.Assertions.assertArrayEquals(java.util.stream.StreamSupport.stream(java.util.Spliterators.spliteratorUnknownSize(#{expected:any(java.util.Iterator<?>)}, 0), false).sorted().toArray(), java.util.stream.StreamSupport.stream(java.util.Spliterators.spliteratorUnknownSize(#{actual:any(java.util.Iterator<?>)}, 0), false).sorted().toArray(), #{message:any(java.lang.String)});")
                        .javaParser(JavaParser.fromJavaVersion().classpath(JavaParser.runtimeClasspath()))
                        .build();
    // …
```

**Comment by timtebeek** [comment]
Thanks for the detailed look & report! I'm not sure if the [RefasterTemplateProcessor](https://github.com/openrewrite/rewrite-templating/blob/7eceffca8aee1e899b2614555dc938083538a20c/src/main/java/org/openrewrite/java/template/processor/RefasterTemplateProcessor.java#L63) would even have access to any formatting or comments (which came up before), but we can certainly strive to do a better job at formatting than placing everything on a single line. Perhaps leveraging auto format in the generated recipes could be of help, but I'm not sure what that would do for performance, or even have the desired effect.

**Comment by knutwannheden** [comment]
Yes, we could certainly add formatting as a default embedding option. That should be easy to implement. Alternatively, maintain the formatting from the template might also be possible. But that would be more work.

**Comment by Philzen** [comment]
> Alternatively, maintain the formatting from the template might also be possible.

This is actually what i'm asking for here. The template string in the generated JavaTemplate.builder should equal what is defined in the `@AfterTemplate` – otherwise you'll get a suboptimal DX, because the accompanying test one writes will fail, with a result that looks non-deterministic to recipe developers.

**Comment by Philzen** [comment]
> Yes, we could certainly add formatting as a default embedding option. 

@knutwannheden There already seems to be some kind of auto-formatting at work, i.e. when using statement lambdas instead of expression lambdas – and the mileage of it varies.

---

Here is an example where it is really helpful as there is literally no way define that kind of formatting as part of the template:

```java
@AfterTemplate void after(Class<? extends Throwable> throwableClass, Executable executable) {
    Assertions.assertThrows(throwableClass, executable);
}
```

… will lead to this migration:

```java
@Test void runnableWithNoExpectedType() {
   // language=java
    rewriteRun(java(
        """
        import org.testng.Assert;
        
        class MyTest {
            void testMethod() {
                Assert.assertThrows(() -> { throw new RuntimeException(); });
            }
        }
        """,
        """
        import org.junit.jupiter.api.Assertions;
        
        class MyTest {
            void testMethod() {
                Assertions.assertThrows(Throwable.class, () -> {
                    throw new RuntimeException();
                });
            }
        }
        """
   ));
}
```

Neat.

---

But [over here](https://github.com/openrewrite/rewrite-templating/issues/90#issuecomment-2180438635) is an example where it very much gets into the way. In that example it becomes really hard to tell that there are two arguments passed to the method as the end of the first one and the beginning of the second one are put on the same line.

---

So again, my suggestion would be to aim at [preserving comments](https://github.com/openrewrite/rewrite-templating/issues/88) and formatting from the `@AfterTemplate` definition as close as possible and not to apply any kind of auto-formatting. 

I was indeed already contemplating if i should post a new issue with an RFC to implement a new annotation for Refaster templates that would enable to configure exactly that, which i would want to add to the refaster template in the first example here.
